packages/edit-site/src/hooks/push-changes-to-global-styles/index.js

```diff
@@ -148,8 +148,12 @@ function PushChangesToGlobalStylesControl( {
 		const newUserConfig = cloneDeep( userConfig );
 
 		for ( const { path, value } of changes ) {
-			set( newBlockStyles, path, undefined );
-			set( newUserConfig, [ 'styles', 'blocks', name, ...path ], value );
+			setImmutably( newBlockStyles, path, undefined );
+			setImmutably(
+				newUserConfig,
+				[ 'styles', 'blocks', name, ...path ],
+				value
+			);
```

Comment:
Unless I'm missing something, this shouldn't work. Unlike `lodash#set`, our helper returns a new object instead of updating the passed argument.

I think return values should be assigned as new config values.

Reply:
@Mamaduka if you have a chance, take a look at the latest implementation.

We're duplicating `setNestedValue` and I'm going to follow-up with deduplicating it afterward.